app/src/main/java/com/weather/weathercheckerapp/MainActivity.kt

```diff
@@ -2,11 +2,36 @@ package com.weather.weathercheckerapp
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProviders
+import com.weather.weathercheckerapp.component.AppComponent
+import com.weather.weathercheckerapp.component.DaggerAppComponent
+import com.weather.weathercheckerapp.viewmodel.MainViewModel
+import javax.inject.Inject
 
 class MainActivity : AppCompatActivity() {
```

Comment:
It would be better to use the Dagger-android features. You can extend `DaggerAppCompatActivity` here and put the DI initialization to a class that extends the Application class.